pkg/resources/collector/workloads.go

```diff
@@ -225,6 +232,15 @@ func (r *Reconciler) daemonSet() resources.Resource {
 				},
 			},
 		},
+		{
+			Name: "filestorage-extension",
+			VolumeSource: corev1.VolumeSource{
+				HostPath: &corev1.HostPathVolumeSource{
+					Path: otelFileStorageDir,
+					Type: &directoryOrCreate,
```

Comment:
This definitely *seems* like the wrong use of of `HostVolume`.
This has the potential to directly affect the underlying host's storage / IO. IMO this is the kind of volatile side effect we would want to avoid especially for agents. Not to mention we want to target low resource environments

We don't have a good way to persist data in the agents at the moment ... Perhaps the way forward is to do something like having the core collector CRD have a field similar to what Opster does where we can specify PVCs to attach and default to something like `EmptyDir` if none are provided

Reply:
Just to note this immediately brought down my node on a small downstream agent that was having sending queue issues while the gateway was restarting

Reply:
Yeah, I agree the current way is potentially dangerous. I'm not familiar with the way Opster does, I'll take a look.

Can you give more details about how the node was brought down? Did it happen on the first installation?

Reply:
Happened on restart, when updating the image

Reply:
we synced up with Dan and this just tracks offsets so this should be fine

Reply:
> Happened on restart when updating the image => the gateway is restarting => gateway is down => agent collector queue filled up

I'm guessing this PR caused the `collector-agents` to restart to pick up the new extension config. During restart, it probably found all the old log files but found no respective offset files. So, the collector-agents interpreted the old and full log files as "new" and began reading them from the start, causing the queues to be full.

This PR literally fixes this bad behavior but also causes it for the very last time. The last dance™.

Reply:
Not sure if we can do anything to make the upgrade run more smoothly. I'll see if we can use `start_at: end` to prevent that.

Reply:
I think that's fine